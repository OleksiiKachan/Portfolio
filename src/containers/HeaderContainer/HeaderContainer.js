import React, { Component } from 'react';
import Header from '../../components/Header/Header.js'

/*
    <HeaderContainer name='' title=''/>
*/
class HeaderContainer extends Component {
    render() {
        return (
            <Header name={this.props.name} title={this.props.title} className={this.props.className} />
        );
    }
}

export default HeaderContainer;
