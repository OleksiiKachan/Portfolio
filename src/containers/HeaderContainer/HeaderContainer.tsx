import React, { Component } from 'react';
import Header from '../../components/Header';

/*
    <HeaderContainer />
*/

export default class HeaderContainer extends Component<{}> {
  private navigationItems = [
    {
      key: 'mi_about',
      title: 'About Me',
      sectionUrl: '/about',
      isSelected: true,
    },
    {
      key: 'mi_projects',
      title: 'Projects',
      sectionUrl: '/projects',
      isSelected: false,
    },
    {
      key: 'mi_competencies',
      title: 'Competencies',
      sectionUrl: '/competencies',
      isSelected: false,
    },
    {
      key: 'mi_education',
      title: 'Education & Experience',
      sectionUrl: '/experience',
      isSelected: false,
    },
    {
      key: 'mi_contact',
      title: 'Contact',
      sectionUrl: '/contact',
      isSelected: false,
    },
  ];

  onLinkClick = (selectedItemKey: string) => {
    (document.getElementById(
      'navigation__input'
    )! as HTMLInputElement).checked = false;

    this.navigationItems = this.navigationItems.map(item => {
      return item.key === selectedItemKey
        ? { ...item, isSelected: true }
        : { ...item, isSelected: false };
    });
  };

  render() {
    return (
      <Header
        onLinkClick={this.onLinkClick}
        navigationItems={this.navigationItems}
      />
    );
  }
}
