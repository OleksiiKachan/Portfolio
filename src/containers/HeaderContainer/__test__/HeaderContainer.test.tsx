import React from 'react';
import { shallow } from 'enzyme';
import HeaderContainer from '..';

describe('HeaderContainer', () => {
  it('Snapshot', () => {
    const component = shallow(<HeaderContainer />);
    expect(component).toMatchSnapshot();
  });
});
