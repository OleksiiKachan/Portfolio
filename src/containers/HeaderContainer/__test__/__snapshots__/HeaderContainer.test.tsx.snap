// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderContainer Snapshot 1`] = `
<Header
  navigationItems={
    Array [
      Object {
        "isSelected": true,
        "key": "mi_about",
        "sectionUrl": "/about",
        "title": "About Me",
      },
      Object {
        "isSelected": false,
        "key": "mi_projects",
        "sectionUrl": "/projects",
        "title": "Projects",
      },
      Object {
        "isSelected": false,
        "key": "mi_competencies",
        "sectionUrl": "/competencies",
        "title": "Competencies",
      },
      Object {
        "isSelected": false,
        "key": "mi_education",
        "sectionUrl": "/experience",
        "title": "Education & Experience",
      },
      Object {
        "isSelected": false,
        "key": "mi_contact",
        "sectionUrl": "/contact",
        "title": "Contact",
      },
    ]
  }
  onLinkClick={[Function]}
/>
`;
